packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -114,12 +121,23 @@ class AnalyzeCommand extends FlutterCommand {
     argParser.addFlag('current-package', help: 'Include the lib/main.dart file from the current directory, if any.', defaultsTo: true);
     argParser.addFlag('preamble', help: 'Display the number of files that will be analyzed.', defaultsTo: true);
     argParser.addFlag('congratulate', help: 'Show output even when there are no errors, warnings, hints, or lints.', defaultsTo: true);
+    argParser.addFlag('watch', help: 'Run analysis continuously, watching the filesystem for changes.', negatable: false);
   }
 
   bool get requiresProjectRoot => false;
 
+  bool get isFlutterRepo {
+    return FileSystemEntity.isDirectorySync('examples') &&
+      FileSystemEntity.isDirectorySync('packages') &&
+      FileSystemEntity.isFileSync('bin/flutter');
+  }
+
   @override
   Future<int> runInProject() async {
+    return argResults['watch'] ? _analyzeWatch() : _analyzeOnce();
+  }
+
+  Future<int> _analyzeOnce() async {
```

Comment:
We could switch over. It will be slower to use the analysis server for one-shot analysis in a few cases. Actually, just for the flutter repo case. Not sure if it's worth paying the 2x slowdown for places where we do need the batch mode, like on CI systems.